addons/netfox.extras/weapon/network-weapon.gd

```diff
@@ -130,39 +132,54 @@ func _before_tick_loop():
 
 	_reconcile_buffer.clear()
 
-func _generate_id(length: int = 12, charset: String = "abcdefghijklmnopqrstuvwxyz0123456789") -> String:
-	var result = ""
-	for i in range(length):
-		var idx = randi_range(0, charset.length() - 1)
-		result += charset[idx]
-	return result
+func _generate_id() -> int:
+	#First projectile
+	if (latest_projectile_id == -1):
+		latest_projectile_id = get_player_id() * 10000
+	else:
+		latest_projectile_id += 1
+	return latest_projectile_id
+
+## @experimental
+## Normally, there would be a singleton which stores all players' peer_ids and player_ids, and the local peer id and player_id
```

Comment:
This also assumes a bit much about the addon user's game; see my comment on get_player_id.